magenta-lib/src/main/scala/magenta/tasks/ASGTasks.scala

```diff
@@ -35,6 +35,23 @@ case class TagCurrentInstancesWithTerminationTag(pkg: DeploymentPackage, stage:
   lazy val description = "Tag existing instances of the auto-scaling group for termination"
 }
 
+case class ProtectCurrentInstances(pkg: DeploymentPackage, stage: Stage, stack: Stack, region: Region)(implicit val keyRing: KeyRing) extends ASGTask {
+  override def execute(asg: AutoScalingGroup, reporter: DeployReporter, stopFlag: => Boolean, asgClient: AmazonAutoScaling) {
+    if (asg.getInstances.asScala.nonEmpty) {
+      val instanceIds = asg.getInstances.asScala.toList.map(_.getInstanceId)
+      val request = new SetInstanceProtectionRequest()
+        .withAutoScalingGroupName(asg.getAutoScalingGroupName)
+        .withInstanceIds(instanceIds: _*)
```

Comment:
Ha, the question indeed: do you need to supply instance IDs?

Reply:
I believe so. The docs say this `should be one or more instance IDs` and I am happy to believe them.

Reply:
Yeah, it is unfortunate the API forces the whole conversion ceremony on you, but it wasn't written with Scala in mind.

Reply:
Any time you interact with a Java API it's the same. The splat operator is actually quite nice in this regard as it skips the need to do a asJava conversion...